apps/omg/lib/omg/state.ex

```diff
@@ -172,53 +170,47 @@ defmodule OMG.State do
   end
 
   @doc """
-  Works exactly like handle_cast(:form_block) but is synchronous
-
-  Also, eth_height given is the Ethereum chain height where the block being closed got submitted, to be used with events.
+  Works exactly like handle_cast(:form_block) but:
+   - is synchronous
+   - `eth_height` given is the Ethereum chain height where the block being closed got submitted, to be used with events.
+   - relies on the caller to handle persistence, instead of handling itself
 
   Someday, one might want to skip some of computations done (like calculating the root hash, which is scrapped)
-
-  Returns `db_updates` due and relies on the caller to do persistence
   """
   def handle_call({:close_block, eth_height}, _from, state) do
-    {:ok, {_block, event_triggers, db_updates}, new_state} = do_form_block(state, eth_height)
-
-    EventerAPI.emit_events(event_triggers)
+    {:ok, {block, event_triggers, db_updates}, new_state} = do_form_block(state, eth_height)
 
+    publish_block_to_event_bus(block, event_triggers)
     {:reply, {:ok, db_updates}, new_state}
   end
 
   @doc """
   Wraps up accumulated transactions submissions into a block, triggers db update and:
-   - emits events to Eventer (if it is running, i.e. in Watcher).
-   - pushes the new block into the respective service (if it is running, i.e. in Child Chain server)
-   - enqueues the new block for submission to BlockQueue (if it is running, i.e. in Child Chain server)
+   - pushes events to subscribers of `"event_triggers"` internal event bus topic
+   - pushes the new block to subscribers of `"blocks"` internal event bus topic
 
   Does its on persistence!
   """
   def handle_cast(:form_block, state) do
     _ = Logger.debug("Forming new block...")
-
-    {duration, {:ok, {%Block{number: blknum} = block, _events, db_updates}, new_state}} =
-      :timer.tc(fn -> do_form_block(state) end)
-
-    _ =
-      Logger.info(
-        "Calculations for forming block number #{inspect(blknum)} done in #{inspect(round(duration / 1000))} ms"
-      )
+    {:ok, {%Block{number: blknum} = block, event_triggers, db_updates}, new_state} = do_form_block(state)
+    _ = Logger.debug("Formed new block ##{blknum}")
 
     # persistence is required to be here, since propagating the block onwards requires restartability including the
     # new block
     :ok = DB.multi_update(db_updates)
 
-    ### casts, note these are no-ops if given processes are turned off
-    BlockQueueAPI.enqueue_block(block)
-
+    publish_block_to_event_bus(block, event_triggers)
     {:noreply, new_state}
   end
 
   defp do_form_block(state, eth_height \\ nil) do
     {:ok, child_block_interval} = Eth.RootChain.get_child_block_interval()
     Core.form_block(child_block_interval, eth_height, state)
   end
+
+  defp publish_block_to_event_bus(block, event_triggers) do
+    :ok = OMG.InternalEventBus.broadcast("events", {:emit_events, event_triggers})
```

Comment:
Do we know enough about scaling to N nodes that we can decide whether we want to broadcast to all nodes in the cluster or use `direct_broadcast`?
Perhaps we don't...

Reply:
I think we don't, but I think `direct_broadcast` is safer for `enqueue_block`, whereas `broadcast` for `events` - intuitively speaking at least. So maybe as future proofing let's do like that